src/features/commands.ts

```diff
@@ -120,58 +114,40 @@ function projectsToCommands(projects: protocol.ProjectDescriptor[], eventStream:
     });
 }
 
-export async function dotnetRestoreAllProjects(server: OmniSharpServer, eventStream: EventStream): Promise<void> {
-
-    if (!server.isRunning()) {
-        return Promise.reject('OmniSharp server is not running.');
+async function pickProjectAndDotnetRestore(server: OmniSharpServer, eventStream: EventStream): Promise<void> {
+    let descriptors = await getProjectDescriptors(server);
+    eventStream.post(new CommandDotNetRestoreStart());
+    let commands = await Promise.all(projectsToCommands(descriptors, eventStream));
+    let command = await vscode.window.showQuickPick(commands);
+    if (command) {
+        return command.execute();
     }
-
-    return serverUtils.requestWorkspaceInformation(server).then(async info => {
-
-        let descriptors = protocol.getDotNetCoreProjectDescriptors(info);
-
-        if (descriptors.length === 0) {
-            return Promise.reject("No .NET Core projects found");
-        }
-
-        let commandPromises = projectsToCommands(descriptors, eventStream);
-
-        return Promise.all(commandPromises).then(commands => {
-            return vscode.window.showQuickPick(commands);
-        }).then(command => {
-            if (command) {
-                return command.execute();
-            }
-        });
-    });
 }
 
-export async function dotnetRestoreForProject(server: OmniSharpServer, filePath: string, eventStream: EventStream) {
+async function dotnetRestoreAllProjects(server: OmniSharpServer, eventStream: EventStream): Promise<void> {
+    let descriptors = await getProjectDescriptors(server);
+    eventStream.post(new CommandDotNetRestoreStart());
+    for (let descriptor of descriptors) {
+        await dotnetRestore(descriptor.Directory, eventStream);
```

Comment:
What does the log output look like while we restore multiple projects?

Reply:
It shows the logs from the restore of each project
![restore](https://user-images.githubusercontent.com/12554141/40671173-cb864e70-6388-11e8-99bc-6b67b1065e35.PNG)